api/v1/controllers/samples.js

```diff
@@ -23,6 +23,45 @@ const doPut = require('../helpers/verbs/doPut');
 const u = require('../helpers/verbs/utils');
 const httpStatus = require('../constants').httpStatus;
 const logAPI = require('../../../utils/loggingUtil').logAPI;
+const WorkerLog = require('../../../utils/activityLogs/workerLog');
+const jwtUtil = require('../../../utils/jwtUtil');
+
+/**
+ * Logs worker activity
+ * @param  {Object} reqNext - The request object and The next middleware
+ * function in the stack
+ * @param  {Object} jobDetails - Worker job and job type
+ * @param  {Integer} reqStartTime - Time in ms when req is received
+ */
+function logWorkerActivity(reqNext, jobDetails, reqStartTime) {
+  // create worker activity log object
+  const workerLogObj = new WorkerLog(jobDetails.jobType.BULKUPSERTSAMPLES);
+  u.getUserNameFromToken(reqNext.req, true)
+  .then((username) => {
+    if (username) {
+      workerLogObj.user = username;
+    }
+  })
+  .then(() => jwtUtil.getTokenNameFromToken(reqNext.req))
+  .then((tokenName) => {
+    workerLogObj.token = tokenName;
+  })
+  .then(() => {
+    workerLogObj.ipAddress = u.getIPAddrFromReq(reqNext.req);
+  })
+  .then(() => {
+    jobDetails.job.on('complete', (result) => {
+      workerLogObj.totalTime = Date.now() - reqStartTime;
+      workerLogObj.queueTime = result.queueTime;
+      workerLogObj.workTime = result.workTime;
+      workerLogObj.dbTime = result.dbTime;
+      workerLogObj.recordCount = result.recordCount;
+      workerLogObj.errorCount = result.errorCount;
+      workerLogObj.toLogString();
+    });
+  })
+  .catch((err) => u.handleError(reqNext.next, err, helper.modelName));
+}
 
```

Comment:
@shriramshankar I like this idea. @pallavi2209 I'll leave it up to you to either add this in this story or create a new technical debt story to do this refactoring.